packages/appium/lib/appium.js

```diff
@@ -241,6 +251,261 @@ class AppiumDriver extends DriverCore {
     }
   }
 
+  /**
+   * Initialize a new bidi connection
+   * @param {import('ws').WebSocket} ws The websocket connection object
+   * @param {import('http').IncomingMessage} req The connection pathname, which might include the session id
+   */
+  async onBidiConnection(ws, req) {
+    ws.on('error', (err) => {
+      // Can't do much with random errors on the connection other than log them
+      this.log.error(err);
+    });
+
+    try {
+      const pathname = req.url;
+      if (!pathname) {
+        throw new Error('Invalid connection request');
+      }
+      const bidiSessionRe = new RegExp(`${BIDI_BASE_PATH}/([^/]+)$`);
```

Comment:
should these patterns be global constants?

Reply:
since they're only used here, i prefer to keep them in their local context. if we ever use them in more than one place, we can factor them out.